src/app/loadout-builder/item-filter.ts

```diff
@@ -94,14 +97,45 @@ export function filterItems(
 function matchedLockedModEnergy(
   defs: D2ManifestDefinitions,
   item: DimItem,
-  lockedModsByPlugCategoryHash: PluggableInventoryItemDefinition[],
+  lockedMods: PluggableInventoryItemDefinition[] | undefined,
   upgradeSpendTier: UpgradeSpendTier,
   lockItemEnergyType: boolean
 ) {
-  if (!lockedModsByPlugCategoryHash) {
+  if (!lockedMods) {
     return true;
   }
-  return lockedModsByPlugCategoryHash.every((mod) =>
+  return lockedMods.every((mod) =>
     doEnergiesMatch(defs, mod, item, upgradeSpendTier, lockItemEnergyType)
   );
 }
+
+function hasEnoughSocketsForMods(
+  defs: D2ManifestDefinitions,
+  item: DimItem,
+  lockedMods: PluggableInventoryItemDefinition[]
+) {
+  if (!lockedMods?.length) {
+    return true;
+  }
+
+  const sockets = getSocketsByCategoryHash(item.sockets!, SocketCategoryHashes.ArmorMods);
+  const plugSets = _.compact(
+    sockets.map(
+      (socket) =>
+        socket.socketDefinition.reusablePlugSetHash &&
+        defs.PlugSet.get(socket.socketDefinition.reusablePlugSetHash).reusablePlugItems.map(
+          (plugItem) => plugItem.plugItemHash
+        )
+    )
+  ).sort(compareBy((plugHashes) => plugHashes.length));
+
+  for (const mod of lockedMods) {
+    const plugSetIndex = plugSets.findIndex((set) => set.includes(mod.hash));
+    if (plugSetIndex === -1) {
+      return false;
+    }
+    plugSets.splice(plugSetIndex, 1);
```

Comment:
Is this what this is doing?
1. Get a list of plugsets for each socket
2. For each locked mod, if it can go into one of the sockets, remove that socket from the list.
3. If we ever can't find a socket, we can't fit them all.

Reply:
Yes exactly. Check my previous comment for an explanation. I will write this into the docs for this function.